lib/pure/algorithm.nim

```diff
@@ -556,6 +558,76 @@ proc isSorted*[T](a: openArray[T], order = SortOrder.Ascending): bool =
     assert isSorted(e) == false
   isSorted(a, system.cmp[T], order)
 
+proc merge*[T](
+  x, y: openArray[T], cmp: proc(x, y: T
+): int {.closure.}, order = SortOrder.Ascending): seq[T] {.since: (1, 5, 1).} =
+  ## Merges two sorted `openArray`. `x` and `y` are assumed to be sorted.
```

Comment:
doc comment needs to specify tie behavior, eg:
merge([(1,1)], [(1,2)], (a,b) => a[0] - b[0])
[(1,1), (1,2)] # should be this IMO
vs
[(1,2), (1,1)]

Reply:
Added it to runnableExamples